src/xrGameSpy/GameSpy_BrowsersWrapper.cpp

```diff
@@ -0,0 +1,290 @@
+#include "stdafx.h"
+#include "GameSpy_BrowsersWrapper.h"
+#include "xrCommon/xr_array.h"
+#include "xrCore/Threading/ScopeLock.hpp"
+
+CGSUpdateStatusAccumulator::CGSUpdateStatusAccumulator(GSUpdateStatus s) { Register(s); }
+
+void CGSUpdateStatusAccumulator::Register(GSUpdateStatus s) { accumulator.push_back(s); }
+
+void CGSUpdateStatusAccumulator::Reset(GSUpdateStatus s)
+{
+    ScopeLock sl(&lock);
+    accumulator.clear();
+    Register(s);
+}
+
+GSUpdateStatus CGSUpdateStatusAccumulator::GetOptimistic() const
+{
+    ScopeLock sl(&lock);
+    GSUpdateStatus res = GSUpdateStatus::Unknown;
+    for (auto s : accumulator)
+    {
+        if (s < res)
+            res = s;
+
+        if (res == GSUpdateStatus::Success)
+            break;
+    }
+    return res;
+}
+
+GSUpdateStatus CGSUpdateStatusAccumulator::GetPessimistic() const
+{
+    ScopeLock sl(&lock);
+    GSUpdateStatus res = GSUpdateStatus::Success;
+    for (auto s : accumulator)
+    {
+        if (s > res)
+            res = s;
+
+        if (res == GSUpdateStatus::Unknown)
+            break;
+    }
+    return res;
+}
+
+bool CGSUpdateStatusAccumulator::IsStatusGood(GSUpdateStatus s) { return (s <= GSUpdateStatus::ConnectingToMaster); }
+
+struct SBrowserConfig
+{
+    CGameSpy_Browser::SMasterListConfig cfg;
+    bool active;
+};
+
+static constexpr SBrowserConfig cop_master_bro = {{GAMESPY_GAMENAME, GAMESPY_GAMEKEY}, true};
+static constexpr SBrowserConfig cs_master_bro = {{GAMESPY_GAMENAME_CS, GAMESPY_GAMEKEY_CS}, true};
+static constexpr SBrowserConfig soc_master_bro = {{GAMESPY_GAMENAME_SOC, GAMESPY_GAMEKEY_SOC}, true};
+
+static constexpr xr_array<SBrowserConfig, 3> master_lists = {{cop_master_bro, cs_master_bro, soc_master_bro}};
```

Comment:
It's better to "open" it like that:
```suggestion
static constexpr xr_array<SBrowserConfig, 3> master_lists = {{ cop_master_bro, cs_master_bro, soc_master_bro }};
```

Reply:
Also, why double braces? If VS 2017 does support compilation with single braces, then it's better to remove them.

Reply:
The auto-formatter doesn't think the code with the spaces is better :) However, looks like it's really possible to get rid of the double braces for master_lists array. Hope GPP will not mind.